Samples/Hast.Samples.Consumer/Gui.cs

```diff
@@ -0,0 +1,356 @@
+using Hast.Layer;
+using Hast.Samples.Consumer.Models;
+using Microsoft.Extensions.Logging;
+using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Terminal.Gui;
+
+namespace Hast.Samples.Consumer
+{
+    public class Gui
+    {
+        private readonly Dictionary<string, ConsumerConfiguration> _savedConfigurations;
+        private readonly ListView _propertiesListView = new ListView { CanFocus = true }.Fill();
+
+        private readonly TextField _optionsTextField =
+            new TextField { CanFocus = true, Visible = false }.FillHorizontally();
+        private Action<string> _currentOptionsTextFieldEventHandler;
+
+        private readonly ListView _optionsListView = new ListView { CanFocus = true, Visible = false }.Fill();
+        private Action<object> _currentOptionsListViewEventHandler;
+
+        private ConsumerConfiguration _configuration;
+
+        private Task<Hastlayer> _hastlayerTask;
+        private Task<List<string>> _deviceNamesTask;
+
+        public Gui(Dictionary<string, ConsumerConfiguration> savedConfigurations) =>
+            _savedConfigurations = savedConfigurations;
+
+        public ConsumerConfiguration BuildConfiguration()
+        {
+            _configuration = new ConsumerConfiguration();
+
+            _hastlayerTask = Task.Run(() => (Hastlayer)Hastlayer.Create(new HastlayerConfiguration()));
+            _deviceNamesTask = _hastlayerTask.ContinueWith(hastlayer =>
+                hastlayer
+                    .Result
+                    .GetSupportedDevices()?
+                    .Select(device => device.Name)
+                    .ToList() ?? new List<string>());

```

Comment:
Miért kell külön `Task` a `_hastlayerTask`-nál, és miért sync van használva mindenhol?

Reply:
Ha exception van akkor [loggert onnan kérünk le](https://github.com/Lombiq/Hastlayer-SDK/pull/59/files/09ede70b6fae079ff85fe4186522e268175e7501#diff-1ca35e59259a53ecdebaa919a6f6ba5bc9ab5234e2505d23f37f54a4f116fe18R122), hogy ne kelljen azt is külön konfigurálni.
Azért sync minden, mert maga a library sync és eleinte úgy tűnt minden az is lesz. A task csak arra kell, hogy ne legyen félmásodperces lag amikor kiválasztod a DeviceName opciót és betölti az driver listát. Ez az igény a fejlesztés folyamán elég későn alakult ki és fölöslegesnek éreztem csak ezért refaktorálni az egészet úgy, hogy nem is vagyok benne biztos, hogy észszerűen lehetséges.

Reply:
OK, ezt dokumentáld ide légyszi.